### DISCUSSION TITLE
How To Detect Overlapping Bonds in Drawn Molecule?

### DISCUSSION THREAD

**Discussion by Julen-Lujambio** [rdkit/rdkit#5762] [Q&A] [answered]
I am generating variations of SMILES strings and then drawing them. I want to remove any drawings that have overlapping bonds like shown in the picture below. 
Is there a way to check if bonds are overlapping in a generated mol drawing? 

![image](https://user-images.githubusercontent.com/57200019/201895726-6df95ee8-6109-4aaf-ba6f-4052dca6b259.png)

**Comment by jose-manuel** [comment, marked as answer]
Dear Julen,
I am not aware of such functionality directly implemented within the RDKit, but the ccdutils repo contains a DepictionValidator class that can be used to detect overlapping atom and bonds:
https://github.com/pschmidtke/ccdutils/blob/master/pdbeccdutils/core/depictions.py

I used it in my project to decide what depiction algorithm performed best for each structure:
https://npfc.readthedocs.io/en/latest/ex_preparation.html#depict

I hope this can be helpful.

Cheers,
Jose Manuel

**Reply by Julen-Lujambio** [reply]
Thank you! This is great.
